lib/client/api.go

```diff
@@ -253,6 +262,46 @@ func (tc *TeleportClient) SSH(command string) error {
 		return trace.Wrap(err)
 	}
 	defer nodeClient.Close()
+
+	// proxy local ports (forward incoming connections to remote host ports)
+	for _, fp := range tc.Config.LocalForwardPorts {
+		socket, err := net.Listen("tcp", net.JoinHostPort("127.0.0.1", strconv.Itoa(fp.SrcPort)))
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		runForwarding := func(remoteAddr string) {
+			for {
+				incoming, err := socket.Accept()
+				if err != nil {
+					log.Error(err)
+					break
+				}
+				defer incoming.Close()
+
+				log.Infof("forwarding connection from %v to %v", incoming.RemoteAddr(), remoteAddr)
+				conn, err := nodeClient.Client.Dial("tcp", remoteAddr)
+				if err != nil {
+					log.Error(err)
+				}
+				defer conn.Close()
+
+				doneC := make(chan interface{}, 2)
+				go func() {
+					io.Copy(incoming, conn)
+					doneC <- true
+				}()
+				go func() {
+					io.Copy(conn, incoming)
+					doneC <- true
+				}()
+				<-doneC
```

Comment:
can you close `conn` and `incoming` when you receive at least one `doneC`? this will allow to mitigate following scenarios:

* incoming is closed by remote side
* first goroutine returns
* second goroutine will never detect that because it's blocked on read from conn